app/scripts/background.js

```diff
@@ -67,8 +75,16 @@ if (inTest || process.env.METAMASK_DEBUG) {
   global.metamaskGetState = localStore.get.bind(localStore);
 }
 
-// initialization flow
-initialize().catch(log.error);
+let initialState;
+let initialLangCode;
+
+const initApp = (remotePort) => initialize(remotePort).catch(log.error);
+browser.runtime.onConnect.addListener(initApp);
+
+(async () => {
+  initialState = await loadStateFromPersistence();
+  initialLangCode = await getFirstPreferredLangCode();
+})();
```

Comment:
Event handler at root level is required for mv3, otherwise in mv3 if user open UI very fast the first event to connect is missed out.